linera-service/src/linera.rs

```diff
@@ -1545,6 +1575,61 @@ impl Runnable for Job {
                 debug!("{:?}", certificate);
             }
 
+            Subscribe {
+                subscriber,
+                publisher,
+                channel,
+            } => {
+                let mut chain_client = context.make_chain_client(storage, subscriber);
+                let time_start = Instant::now();
+                info!("Subscribing");
+                let result = match channel {
+                    SystemChannel::Admin => chain_client.subscribe_to_new_committees().await,
+                    SystemChannel::PublishedBytecodes => match publisher {
+                        Some(publisher_chainid) => {
+                            chain_client
+                                .subscribe_to_published_bytecodes(publisher_chainid)
+                                .await
+                        }
+                        None => Err(ChainClientError::InternalError("Incorrect chain ID")),
+                    },
+                };
+                context.update_and_save_wallet(&mut chain_client).await;
+                let subscribe = result.context("Failed to subscribe")?;
+                let time_total = time_start.elapsed().as_micros();
+                info!("Subscription confirmed after {} us", time_total);
```

Comment:
We're in the process of switching to `millis`.

Reply:
Okay 👍